pages/game/[gameId].tsx

```diff
@@ -24,10 +26,22 @@ function GamePage(props: GamePageProps): JSX.Element {
   const { gameId, isHost } = props;
 
   const player = useSelector(selectPlayer);
+  const gameRoom = useSelector(selectGameRoom);
+
+  const session = useAuth();
+
+  useEffect(() => {
+    if (gameRoom || !isHost) return;
+
+    UrlHelpers.changeLocation(PATHS.DEFAULT);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   useEffect(() => {
-    if (!isHost && player) emitPlayerJoiningGame(gameId, player);
-  }, [gameId, isHost, player]);
+    if (session === undefined || isHost || !player) return;
```

Comment:
Unless I am mistaken 
```suggestion
    if (!session || isHost || !player) return;
```